cmd/lotus-shed/f3.go

```diff
@@ -17,6 +28,170 @@ var f3Cmd = &cli.Command{
 	Description: "f3 related commands",
 	Subcommands: []*cli.Command{
 		f3ClearStateCmd,
+		f3GenExplicitPower,
+	},
+}
+
+func loadF3IDList(path string) ([]gpbft.ActorID, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %w", err)
+	}
+	defer file.Close() //nolint:errcheck
+
+	var ids []gpbft.ActorID
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" || line[0] == '#' {
+			continue
+		}
+		id, err := strconv.ParseUint(line, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse ID: %w", err)
+		}
+
+		ids = append(ids, gpbft.ActorID(id))
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error reading file: %w", err)
+	}
+
+	return ids, nil
+}
+
+var f3GenExplicitPower = &cli.Command{
+	Name:        "gen-explicit-power",
+	Description: "generates an explicit power table",
+
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:  "good-list",
+			Usage: "new line delimited file with known good IDs to be included",
+		},
+		&cli.PathFlag{
+			Name:  "bad-list",
+			Usage: "new line delimited file with known bad IDs to be excluded",
+		},
+		&cli.IntFlag{
+			Name:  "N",
```

Comment:
```suggestion
			Name:  "n",
```

how about just lower-case for a CLI flag